salt/utils/minions.py

```diff
@@ -239,9 +239,11 @@ def _pki_minions(self):
         Retreive complete minion list from PKI dir.
         Respects cache if configured
         '''
-        if self.opts.get('__role') == 'master' and self.opts.get('__cli') == 'salt-run':
-            # Compiling pillar directly on the master, just return the master's
-            # ID as that is the only one that is available.
+        opts_role = self.opts.get('__role')
+        if ((opts_role == 'master' and self.opts.get('__cli') == 'salt-run') or
+         (opts_role == 'minion')):
```

Comment:
I think we have an issues with the indent here. Maybe just put it in one line, it anyway reads better then.